apps/addons/models.py

```diff
@@ -528,6 +523,28 @@ def from_upload(cls, upload, platforms, is_packaged=False, source=None,
                          data.get('default_locale') == addon.default_locale)
         if not locale_is_set:
             addon.default_locale = to_language(translation.get_language())
+
+        return addon
+
+    @classmethod
+    def create_addon_from_upload_data(cls, data, is_listed=True, user=None,
+                                      status=amo.STATUS_NULL):
+        addon = cls.initialize_addon_from_upload(
+            data, is_listed=is_listed, status=status)
```

Comment:
if you passed `is_listed` and `status` as `**kw` then you wouldn't have to re-declare their default values.